sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -0,0 +1,742 @@
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Create, read, update and delete items in the Azure Cosmos DB SQL API service.
+"""
+
+from typing import Any, AnyStr, Dict, List, Optional, Union, Iterable, cast
+
+from azure.core.tracing.decorator import distributed_trace  # pylint: disable=unused-import
+from azure.core.tracing.decorator_async import distributed_trace_async  # type: ignore
+from typing import Any, Dict, List, Optional, Union, Iterable, cast  # pylint: disable=unused-import
+
+from ._cosmos_client_connection_async import CosmosClientConnection
+from .._base import build_options
+from ..exceptions import CosmosResourceNotFoundError
+from ..http_constants import StatusCodes
+from ..offer import Offer
+from .scripts import ScriptsProxy
+from ..partition_key import NonePartitionKeyValue
+
+__all__ = ("ContainerProxy",)
+
+# pylint: disable=protected-access
+# pylint: disable=missing-client-constructor-parameter-credential,missing-client-constructor-parameter-kwargs
+
+class ContainerProxy(object):
+    """An interface to interact with a specific DB Container.
+
+    This class should not be instantiated directly. Instead, use the
+    :func:`DatabaseProxy.get_container_client` method to get an existing
+    container, or the :func:`Database.create_container` method to create a
+    new container.
+
+    A container in an Azure Cosmos DB SQL API database is a collection of
+    documents, each of which is represented as an Item.
+
+    :ivar str id: ID (name) of the container
+    :ivar str session_token: The session token for the container.
+    """
+
+    def __init__(self, client_connection, database_link, id, properties=None):  # pylint: disable=redefined-builtin
+        # type: (CosmosClientConnection, str, str, Dict[str, Any]) -> None
+        self.client_connection = client_connection
+        self.id = id
+        self._properties = properties
+        self.container_link = u"{}/colls/{}".format(database_link, self.id)
+        self._is_system_key = None
+        self._scripts = None  # type: Optional[ScriptsProxy]
+
+    def __repr__(self):
+        # type () -> str
+        return "<ContainerProxy [{}]>".format(self.container_link)[:1024]
+
+    async def _get_properties(self):
+        # type: () -> Dict[str, Any]
+        if self._properties is None:
+            self._properties = await self.read()
+        return self._properties
+
+    @property
+    async def is_system_key(self):
```

Comment:
Seems to be used to check if partitionKey is system generated or not - although with that said the SDK makes partition key a required argument in container creation and so does the portal, so this could be removed, would like to know more about "system generated" partition keys before removing though